src/qibolab/instruments/qblox/cluster_qcm_bb.py

```diff
@@ -311,26 +260,16 @@ def _get_next_sequencer(self, port, frequency, qubits: dict):
                     value = self.device.sequencers[self.DEFAULT_SEQUENCERS[port]].get(param_name=parameter)
                     if value:
                         target = self.device.sequencers[next_sequencer_number]
-                        self._set_device_parameter(target, parameter, value=value)
+                        target.set(parameter, value)
 
         # if hardware modulation is enabled configure nco_frequency
         if self.ports[port].hardware_mod_en:
-            self._set_device_parameter(
-                self.device.sequencers[next_sequencer_number],
-                "nco_freq",
-                value=frequency,  # Assumes all pulses in non_overlapping_pulses set
-                # have the same frequency. Non-overlapping pulses of different frequencies on the same
-                # qubit channel with hardware_demod_en would lead to wrong results.
-                # TODO: Throw error in that event or implement for non_overlapping_same_frequency_pulses
-                #       Even better, set the frequency before each pulse is played (would work with hardware modulation only)
-            )
-
-            # set sequencer port offset to the runcard value
-            # self._set_device_parameter(
-            #     self.device.sequencers[next_sequencer_number],
-            #     "out0_offset", #How to select device parameter offset??
-            #     value=qubits[qubit].sweetspot,
-            # )
+            self.device.sequencers[next_sequencer_number].set("nco_freq", frequency)
+            # Assumes all pulses in non_overlapping_pulses set
+            # have the same frequency. Non-overlapping pulses of different frequencies on the same
+            # qubit channel with hardware_demod_en would lead to wrong results.
+            # TODO: Throw error in that event or implement for non_overlapping_same_frequency_pulses
+            # Even better, set the frequency before each pulse is played (would work with hardware modulation only)
```

Comment:
One more chapter in the sw/hw modulation series: maybe the "even better" would be the optimal solution.

It'd be useful to know if we really need to maintain the software modulation.

Reply:
I think none of the other drivers support software modulation so it is not required for any of the things we are usually doing. So the naive answer would be that we don't need to maintain it, however I would ask Alvaro or someone more familiar with lab procedures before dropping it.

Reply:
I spoke this morning with Alvaro and he said that software modulation has some  niche application that he is not sure we will ever use (e.g. sending pulses that are not sinusoidal inside the envelopes but have others arbitrary form). So I think we can drop this option.

Reply:
At the moment, we're not implementing (de)modulation for non-sinusoidal carriers is not implemented, so we're maintaining a duplicate.

Moreover, if the carrier is software modulated, we cannot rely on the
LOs and the mixers, so the frequency would be the one of the DAC as it is, with all its limitations.